# [ DATA SCIENCE ] Assignment
# [ BASIC STATISTICS LEVEL - 2]
# S.AKHIL
# Gmail : akhilsangamrjasn@gmail.com

In [22]:
import numpy as np
from scipy.stats import t, norm

In [24]:
#GIVEN DATA TAKEN IN ARRAY
x = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
y = len(x)
C_I = 0.99 #CONFIDENCE INTERVAL
alpha = 1 - C_I
# alpha = 0.01

In [26]:
# Sample mean and standard deviation
sample_mean = np.mean(x)
sample_std = np.std(x, ddof=1)

## TASK 1: Build a 99% confidence interval using Sample_Standard_Deviation

In [31]:
# Part A: Confidence interval using sample standard deviation (t-distribution)
t_critical = t.ppf(1 - alpha / 2, df=y - 1)
margin_of_error_t = t_critical * (sample_std / np.sqrt(y))
ci_t = (sample_mean - margin_of_error_t, sample_mean + margin_of_error_t)

## TASK 2: Build a 99% confidence interval using known Population_Standard_Deviation

In [33]:
# Part B: Confidence interval using known population standard deviation (z-distribution)
population_std = 0.2
z_critical = norm.ppf(1 - alpha / 2)
margin_of_error_z = z_critical * (population_std / np.sqrt(y))
ci_z = (sample_mean - margin_of_error_z, sample_mean + margin_of_error_z)

sample_mean, sample_std, ci_t, ci_z

(1.2386666666666666,
 0.19316412956959936,
 (1.0901973384384906, 1.3871359948948425),
 (1.1056514133957607, 1.3716819199375725))

In [35]:
print('Sample mean =', sample_mean)
print('sample_standard =', sample_std)
print('t-Critical =', t_critical)
print('Margin_of_Error=', margin_of_error_z)
print('Population_standard_Deviation:',ci_z )

Sample mean = 1.2386666666666666
sample_standard = 0.19316412956959936
t-Critical = 2.976842734370834
Margin_of_Error= 0.13301525327090588
Population_standard_Deviation: (1.1056514133957607, 1.3716819199375725)


# CONCULSION

1. Sample_Mean                    = 1.239 million.
2. Sample_standard_Deviation      = 0.193 million.
3. 99% Confidence Interval        = 1.090 million to 1.372 million
4. Population_Standard_Deviation  = 0.2 million
5. 99% Confidence Interval        = 1.106 million to 1.372 million

The confidence interval using the sample standard deviation is slightly wider than the one using the known population standard ddeviation. This is because estimating the standard deviation introduces additional uncertainty.